Updating issue and PR templates (#3132)
TeamNewPipe/NewPipe: .github/ISSUE_TEMPLATE/feature_request.md

---
name: Feature request
about: Suggest an idea for this project
title: "[Feature]"
labels: enhancement
assignees: ''

---

#### Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is.

> Ex. *I want to do X, but there is no way to do it.*

Author: I think you can just use what you want. If it feels better for you, we can switch to E. g., but Ex. sounds as clear to me.

> Ex. *I want to do X, but there is no way to do it.*

Reviewer: `Ex.` is wrong here semantically. It abbreviates "example". In this case, you have to go for either of @B0pol's suggestions (they're all basically the same). IMO using `for example` is the "simplest", and as we won't win a price for the most elegant resp. complex English sentence, we should go for accessibility. I have met dozens of non-native speakers who couldn't explain `e.g.`, and many of them also have never heard `for instance`.

> Ex. *I want to do X, but there is no way to do it.*

Author: Cool, Example: it is

#### Describe the solution you'd like
A clear and concise description of what you want to happen.
Ex. *I think it would be nice if you would add feature Y so it will make it easier.*

#### Describe alternatives you've considered
A clear and concise description of any alternative solutions or features you've considered.
Ex. *I considered Z, but that didn't work because...*

#### Additional context
Add any other context or screenshots about the feature request here.

> Ex. *Here's a photo of my cat!*

Reviewer 2: We are at the bottom of the template, and I disagree about "a clear and concise description" everywhere.

IMO, at least one paragraph should say:
Why you'd really want this feature? Explain fow it will change your life and your NewPipe experience!

This was just an example, but the idea is to have one paragraph where people THINK about their request. Do they really WANT it, or it was just a thought they had but they don't really care about it.

And if they give good reasons why they really WANT it, we can understand better, and prioritize (dunno if it's how we say) these requests, or on the contrary, if they didn't give good reasons AND it seems like a **niche** request, maintaners / contributors will know there is more important things to fix / features to bring

Author: > and I disagree about "a clear and concise description" everywhere.

You would delete this sentence everywhere?

>Why you'd really want this feature? Explain how it will change your life and your NewPipe experience!

MY DEAD MOTHER COULD COME BACK! yeah I get you good point

Reviewer 2: > You would delete this sentence everywhere?

No, I didn't meant to remove it everywhere. My bad ih you understood that

Author: okay, uhm. how can I understand that part then

---
name: Feature request
about: Suggest an idea for this project
title: "[Feature]"
labels: enhancement
assignees: ''

---

> <!-- Hey. Our [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) might be a nice

Reviewer 3: And here

> <!-- Hey. Our contribution guidelines (https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) might be an approptiate

Reviewer 4: ```suggestion
<!-- Hey. Our contribution guidelines (https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) might be an appropriate
```

#### Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is.
Ex. *I want to do X, but there is no way to do it.*

#### Describe the solution you'd like
A clear and concise description of what you want to happen.
Ex. *I think it would be nice if you would add feature Y so it will make it easier.*

#### Describe alternatives you've considered
A clear and concise description of any alternative solutions or features you've considered.
Ex. *I considered Z, but that didn't work because...*

#### Additional context
Add any other context or screenshots about the feature request here.
Ex. *Here's a photo of my cat!*

#### Why do you/everyone wants this feature
Convince us! How does it change your NewPipe experience and/or your life?

> The better this paragraph is, the more likely a developer will think about developing it

Reviewer 2: ```suggestion
The better this paragraph is, the more likely a developer will think about developing it.
```

Author: I just do it for clearer view in the source code/for the template. Its like a bigger dot. (In this case btw, people should replace it anyway)

---
name: Feature request
about: Suggest an idea for this project
title: "[Feature]"
labels: enhancement
assignees: ''

---
<!-- Hey. Our [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) might be a nice

>  document to read before you fill out the request :) -->

Reviewer 4: ```suggestion
 document to read before you fill out the request. :) -->
```

Author: imo, the emoji replaces the punctuation

> #### Is your feature request related to a problem? Please describe.

Reviewer 4: Consistency with the other headings
```suggestion
#### Is your feature request related to a problem? Please describe
```

A clear and concise description of what the problem is.
Example: *I want to do X, but there is no way to do it.*

#### Describe the solution you'd like
A clear and concise description of what you want to happen.
Example: *I think it would be nice if you would add feature Y so it will make it easier.*

#### Describe alternatives you've considered
A clear and concise description of any alternative solutions or features you've considered.

> Example: *I considered Z, but that didn't work because...*

Reviewer 5: ```suggestion
Example: *I considered Z, but that didn't turn out to be a good idea because...*
```

#### Additional context
Add any other context or screenshots about the feature request here.
Example: *Here's a photo of my cat!*

> #### Why do you/everyone wants this feature

Author: I like benefit. I like you. Thanks

> #### Why do you/everyone wants this feature

Reviewer 4: I benefit from you liking me. Welcome.

Convince us! How does it change your NewPipe experience and/or your life?

> The better this paragraph is, the more likely a developer will think about developing it.

Reviewer 4: ```suggestion
The better this paragraph is, the more likely a developer will think about working on it.
```
